### PR TITLE
[popover2] feat: Popover2 popupkind

### PR DESCRIPTION
**Pull request by dlech**
#### Checklist

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:
This adds a new `popupKind` property to the `Popover2` component for controlling the `aria-haspopup` attribute of the target element.

The current implementation hard-codes `aria-haspopup` to "true" which means that the popup is a menu. However, this is not true in all cases. Furthermore, the `aria-haspopup` attribute should not be set for tooltips since they are not interactive elements. So if the interaction kind is a hover interaction, this new property is ignored.

Reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup

#### Screenshot

Before:
image

After:
image

### CODE REVIEW

File: packages/popover2/src/popover2.tsx [hunk 1]
```diff
@@ -50,14 +50,28 @@ export const Popover2InteractionKind = {
 // eslint-disable-next-line @typescript-eslint/no-redeclare
 export type Popover2InteractionKind = typeof Popover2InteractionKind[keyof typeof Popover2InteractionKind];
 
+/** Specifies the popup kind for aria-haspopup. */
+export enum Popover2PopupKind {
```
Comment (adidahiya) [inline packages/popover2/src/popover2.tsx]: I think we can generalize this enum name and just call it `PopupKind`, and move it to a new file `popupKind.ts` in this package. Also please include a link to the [docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup) in the JSDoc for this symbol.

File: packages/popover2/src/popover2.tsx [hunk 2]
```diff
@@ -50,14 +50,28 @@ export const Popover2InteractionKind = {
...
+
 // eslint-disable-next-line deprecation/deprecation
 export type Popover2Props<TProps = React.HTMLProps<HTMLElement>> = IPopover2Props<TProps>;
 /** @deprecated use Popover2Props */
 export interface IPopover2Props<TProps = React.HTMLProps<HTMLElement>> extends Popover2SharedProps<TProps> {
     /**
      * Whether the popover/tooltip should acquire application focus when it first opens.
      *
-     * @default true for click interations, false for hover interactions
+     * @default true for click interactions, false for hover interactions
```
Comment (adidahiya) [inline packages/popover2/src/popover2.tsx:62]: thanks for fixing the typos :)

File: packages/popover2/src/popover2.tsx [hunk 3]
```diff
@@ -327,7 +352,7 @@ export class Popover2<T> extends AbstractPureComponent2<Popover2Props<T>, IPopov
         // Ensure target is focusable if relevant prop enabled
         const targetTabIndex = openOnTargetFocus && isHoverInteractionKind ? 0 : undefined;
         const targetProps = {
-            "aria-haspopup": "true",
+            "aria-haspopup": this.props.popupKind ?? (isHoverInteractionKind ? undefined : "true"),
```
Comment (adidahiya) [inline packages/popover2/src/popover2.tsx]: I think this should be only `undefined` if `this.props.interactionKind === Popover2InteractionKind.HOVER_TARGET_ONLY`.

in the case of `Popover2InteractionKind.HOVER` popovers, you _can_ still interact with the content, and we have instances of that in our applications. so haspopup should be set to something.

when you make this change, please make sure to update the new prop JSDoc.

File: packages/popover2/test/popover2Tests.tsx [hunk 4]
```diff
@@ -192,6 +192,16 @@ describe("<Popover2>", () => {
...
             assert.isTrue(wrapper.find("[aria-haspopup='true']").exists());
         });
+
+        it("sets aria-haspopup attr base on popupKind", () => {
+            wrapper = renderPopover({ isOpen: true, popupKind: Popover2PopupKind.DIALOG });
+            assert.isTrue(wrapper.find("[aria-haspopup='dialog']").exists());
+        });
+
+        it("renders without aria-haspopup attr for hover interaction", () => {
+            wrapper = renderPopover({ isOpen: true, interactionKind: Popover2InteractionKind.HOVER });
```
Comment (adidahiya) [inline packages/popover2/test/popover2Tests.tsx]: this test will have to be updated if you make my change suggested above.

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: packages/popover2/src/popover2.tsx (hunk 1 above)
Comment (dlech) [reply packages/popover2/src/popover2.tsx]: done

File: packages/popover2/src/popover2.tsx (hunk 3 above)
Comment (dlech) [reply packages/popover2/src/popover2.tsx]: done

File: packages/popover2/test/popover2Tests.tsx (hunk 4 above)
Comment (dlech) [reply packages/popover2/test/popover2Tests.tsx]: done

**Review by adidahiya — APPROVED** [review]

**Comment by adidahiya** [comment]
thanks @dlech!

**Pull request merged**